clients/src/main/java/org/apache/kafka/clients/admin/internals/AlterConsumerGroupOffsetsHandler.java

```diff
@@ -0,0 +1,156 @@
+            });
+        }
+        for (Map.Entry<String, List<OffsetCommitRequestPartition>> entry : offsetData.entrySet()) {
+            OffsetCommitRequestTopic topic = new OffsetCommitRequestTopic()
+                    .setName(entry.getKey())
+                    .setPartitions(entry.getValue());
+            topics.add(topic);
+        }
+        OffsetCommitRequestData data = new OffsetCommitRequestData()
+            .setGroupId(groupId.idValue)
+            .setTopics(topics);
+        return new OffsetCommitRequest.Builder(data);
+    }
+
+    @Override
+    public ApiResult<CoordinatorKey, Map<TopicPartition, Errors>> handleResponse(Node broker, Set<CoordinatorKey> groupIds,
+            AbstractResponse abstractResponse) {
+
+        final OffsetCommitResponse response = (OffsetCommitResponse) abstractResponse;
+        Map<CoordinatorKey, Map<TopicPartition, Errors>> completed = new HashMap<>();
+        Map<CoordinatorKey, Throwable> failed = new HashMap<>();
+        List<CoordinatorKey> unmapped = new ArrayList<>();
+
+        Map<TopicPartition, Errors> partitions = new HashMap<>();
+        for (OffsetCommitResponseTopic topic : response.data().topics()) {
+            for (OffsetCommitResponsePartition partition : topic.partitions()) {
+                TopicPartition tp = new TopicPartition(topic.name(), partition.partitionIndex());
+                Errors error = Errors.forCode(partition.errorCode());
+                if (error != Errors.NONE) {
+                    handleError(groupId, error, failed, unmapped);
+                } else {
+                    partitions.put(tp, error);
+                }
+            }
+        }
+        if (failed.isEmpty() && unmapped.isEmpty())
+            completed.put(groupId, partitions);
+
+        return new ApiResult<>(completed, failed, unmapped);
+    }
+
+    private void handleError(CoordinatorKey groupId, Errors error, Map<CoordinatorKey, Throwable> failed, List<CoordinatorKey> unmapped) {
+        switch (error) {
+            case GROUP_AUTHORIZATION_FAILED:
+                log.error("Received authorization failure for group {} in `DeleteConsumerGroupOffsets` response", groupId,
+                        error.exception());
+                failed.put(groupId, error.exception());
+                break;
+            case COORDINATOR_LOAD_IN_PROGRESS:
+            case COORDINATOR_NOT_AVAILABLE:
+            case NOT_COORDINATOR:
+                log.debug("DeleteConsumerGroupOffsets request for group {} returned error {}. Will retry", groupId, error);
+                unmapped.add(groupId);
+                break;
+            default:
+                log.error("Received unexpected error for group {} in `DeleteConsumerGroupOffsets` response", 
+                        groupId, error.exception());
+                failed.put(groupId, error.exception(
+                        "Unexpected error during DeleteConsumerGroupOffsets lookup for " + groupId));
```

Comment:
Yes some of these handlers different in pretty subtle ways. Unfortunately if we remove this `break` statement, some of the existing tests stop working. In this PR, I've aimed at not touching the exiting tests to ensure my refactoring don't change the current behaviour.

I agree, this is very confusing and I think we should align behaviour for all these similar calls and have a generic error handling logic. But I would prefer to not do it as part of this PR as it's already pretty big.

Reply:
@mimaison makes sense, it would be good to file a JIRA for this so we can address it in a future PR to ensure we have consistent error handling across all consumer group related issues.

Reply:
I agree, I raised https://issues.apache.org/jira/browse/KAFKA-13012